packages/blocks/src/api/utils.js

```diff
@@ -121,3 +123,105 @@ export function normalizeBlockType( blockTypeOrName ) {
 
 	return blockTypeOrName;
 }
+
+/**
+ * Get the label for the block, usually this is either the block title,
+ * or the value of the block's `label` function when that's specified.
+ *
+ * @param {Object} blockType  The block type.
+ * @param {Object} attributes The values of the block's attributes.
+ * @param {Object} context    The intended use for the label.
+ *
+ * @return {string} The block label.
+ */
+export function getBlockLabel( blockType, attributes, context = 'visual' ) {
+	const {
+		__experimentalLabel: getLabel,
+		title,
+	} = blockType;
+
+	const label = getLabel && getLabel( attributes, { context } );
+
+	if ( ! label ) {
+		return title;
+	}
+
+	// Strip any HTML (i.e. RichText formatting) before returning.
+	return stripHTML( label );
```

Comment:
Can't we make this the responsibility of the block implementation in generating the label? In other words, force the block to return a plain string. There, we already have some availability of the `rich-text` tooling. I see there was some prior discussion about `rich-text` being used here, but if we would put it on the block to return, then it doesn't need to be a dependency here.

I'm not sure there exists a utility already for generating a plain-text string from a RichText value, but it would seem to align with some of the existing utilities (`toTree`, `toDOM`, `toHTMLString`). Something like a `toPlainString` (cc @ellatrix).

I am a bit worried about introducing something like a `stripHTML`, both for these reasons above in that it ideally should not be necessary, but also because it could become a security weakness† or a source of poor performance.

_† Not to say that it's vulnerable today in how it's used, notably because we're not `dangerouslySetInnerHTML`, but it's one of those things where we should try to avoid this altogether if we can, for fear of future mis-use, browser inconsistencies, etc._

Reply:
So I had some feedback that the `__experimentalLabel` api on the block needed some thought. It looks like the option is to split it into two separate things:

This for the accessibility label:
```
__experimentalGetAccessibilityLabel( attributes ) { return attributes.title }
```

and this for the label that might appear in the block navigator:
```
__experimentalLabel: 'title' // title is the name of the attribute
```

So it looks like the option of moving this responsibility to the block might not be possible for the latter. One option is something that @youknowriad mentioned in another comment, is that these could become selectors in the `block-editor` package. I started working on that, and I think it works for the accessibility label. For the visual label it would be a selector that doesn't select anything from the store, which is a little unusual, but it might be a pragmatic option to have the function in the selectors file.

Then I could change `stripHTML` to `toPlainString` as suggested.